### PR TITLE
Refactor Chart classes

### PR DESCRIPTION
**Pull request by kinegratii**
See #768 for detail.

### CODE REVIEW

**Comment by chenjiandongx** [comment]
@kinegratii 如果 `Style`  类要废弃的话 http://pyecharts.org/#/zh-cn/charts_style 这里可以多介绍一点

File: docs/zh-cn/technical.md [hunk 1]
```diff
@@ -10,6 +10,13 @@
 - [black(18.6b4)](https://github.com/ambv/black) ：代码格式化
 - [McCabe](https://pypi.org/project/mccabe/) ：复杂度检测工具
 
+## 开发约定
+
+- 每行不超过 79 个字符。
+- 函数复杂度不超过 14 。
+- 代码覆盖率不低于 90% 。
+- `# flake8: noqa` 应用于整个文件，应当放置在文件开始的部分。
```
Comment (chfw) [inline docs/zh-cn/technical.md]: # flake8 on top will silence the whole file. 

sometimes, it is required to append to a line so as to silence the warning on the line itself. but I recommend that do not use "flake8: noqa" in general. For our __init__.py, it is used but it is an exception.

File: pyecharts/interfaces.py [hunk 2]
```diff
@@ -12,3 +13,21 @@ class JSONSerializableMixin(object):
 
     def __json__(self):
         pass
+
+
+class IPythonRichDisplayMixin(object):
```
Comment (chfw) [inline pyecharts/interfaces.py]: Well, it is OK to have explicit interface.

File: docs/zh-cn/charts_style.md [hunk 3]
```diff
@@ -1,5 +1,7 @@
 > 图表风格篇：本篇文档为 pyecharts 图表风格详情文档。
 
+> 重要：从 v0.6.0 开始，类 Style 已经废弃。可以使用原生 dict 代替。
```
Comment (chenjiandongx) [inline docs/zh-cn/charts_style.md]: 要不给个小 demo 吧

Comment (kinegratii) [reply docs/zh-cn/charts_style.md]: 只是简单的变量复用，无需另行说明，

**Pull request merged**